go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/sourcenetwork/defradb
 
-go 1.21.3
+go 1.21.5
+
+toolchain go1.22.2
```

Comment:
todo: Why has this been added?

Reply:
Honestly, no idea. `go mod tidy` changed it. I'm not too familiar with the purpose of the `toolchain` directive. Has this happened before? I can look into or manually remove it

Reply:
Ah fair enough, I'd suggest just reverting this and the `go` version then :)

Reply:
I removed the toolchain directive, which fixed the vulnerability workflow. I left the minor bump as is, however I can revert it as well if that's preferred

Reply:
I would prefer that we don't bump the minor version in random PRs.  Although I'm not really sure why we even specify the minor version.

Reply:
> I would prefer that we don't bump the minor version in random PRs. Although I'm not really sure why we even specify the minor version - might be a historical hangover that we havent yet removed from a previous version where we really wanted a minor version feature.

+1, also our `mod tidy` make rule (`make tidy`) ensures the modules are "clean" up to a minimum version. This was due to some pains that were encountered when a `mod tidy` command breaks and expects a minimum patch version. For example when the commit `c42e7ee6 feat: Update corelog to 0.0.7 (#2547)` made to develop it broke the go mod file. 

tldr: if you want to bump the version it should also update the `make tidy` Makefile rule and I would highly suggest to do this in a separate PR if you don't need the version bump here.

This reminds me, we should add a ci check to ensure our `make tidy` is never broken, made issue here: https://github.com/sourcenetwork/defradb/issues/2695

Reply:
:saluting_face: